qiskit/aqua/algorithms/phase_estimators/hamiltonian_pe.py

```diff
@@ -0,0 +1,174 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Phase estimation for the spectrum of a Hamiltonian"""
+
+from typing import Optional, Union
+import numpy
+from qiskit import QuantumCircuit
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.operators import EvolutionBase, OperatorBase
+from qiskit.providers import BaseBackend
+
+# TODO: Remove temporary code when possible.
```

Comment:
How are we on the fixes etc. Can any of this be updated - or is it better to flag this PR as on hold pending whatever is going on.